diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4042f001b..2e49740b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -657,9 +657,9 @@ macro(openssl_check_quic)
       openssl_check_symbol_exists(wolfSSL_set_quic_method "wolfssl/options.h;wolfssl/openssl/ssl.h" HAVE_SSL_CTX_SET_QUIC_METHOD)
     endif()
   endif()
-  if(NOT HAVE_SSL_CTX_SET_QUIC_METHOD)
-    message(FATAL_ERROR "QUIC support is missing in OpenSSL fork. Try setting -DOPENSSL_ROOT_DIR")
-  endif()
+  # if(NOT HAVE_SSL_CTX_SET_QUIC_METHOD)
+  #   message(FATAL_ERROR "QUIC support is missing in OpenSSL fork. Try setting -DOPENSSL_ROOT_DIR")
+  # endif()
 endmacro()
 
 if(USE_OPENSSL OR USE_WOLFSSL)
diff --git a/src/curl.rc b/src/curl.rc
index 6fcaf353e..24f33afcb 100644
--- a/src/curl.rc
+++ b/src/curl.rc
@@ -26,6 +26,8 @@
 
 LANGUAGE  0, 0
 
+ICONAPP ICON "curl.ico"
+
 #define RC_VERSION  CURL_VERSION_MAJOR, CURL_VERSION_MINOR, CURL_VERSION_PATCH, 0
 
 VS_VERSION_INFO VERSIONINFO
