diff --git a/CMakeLists.txt b/CMakeLists.txt
index 45d763d5a..6604aa74e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -508,23 +508,23 @@ if(USE_NGHTTP2)
   list(APPEND CURL_LIBS ${NGHTTP2_LIBRARIES})
 endif()
 
-function(CheckQuicSupportInOpenSSL)
-  # Be sure that the OpenSSL library actually supports QUIC.
-  cmake_push_check_state()
-  set(CMAKE_REQUIRED_INCLUDES   "${OPENSSL_INCLUDE_DIR}")
-  set(CMAKE_REQUIRED_LIBRARIES  "${OPENSSL_LIBRARIES}")
-  check_symbol_exists(SSL_CTX_set_quic_method "openssl/ssl.h" HAVE_SSL_CTX_SET_QUIC_METHOD)
-  if(NOT HAVE_SSL_CTX_SET_QUIC_METHOD)
-    message(FATAL_ERROR "QUIC support is missing in OpenSSL/boringssl. Try setting -DOPENSSL_ROOT_DIR")
-  endif()
-  cmake_pop_check_state()
-endfunction()
+# function(CheckQuicSupportInOpenSSL)
+#   # Be sure that the OpenSSL library actually supports QUIC.
+#   cmake_push_check_state()
+#   set(CMAKE_REQUIRED_INCLUDES   "${OPENSSL_INCLUDE_DIR}")
+#   set(CMAKE_REQUIRED_LIBRARIES  "${OPENSSL_LIBRARIES}")
+#   check_symbol_exists(SSL_CTX_set_quic_method "openssl/ssl.h" HAVE_SSL_CTX_SET_QUIC_METHOD)
+#   if(NOT HAVE_SSL_CTX_SET_QUIC_METHOD)
+#     message(FATAL_ERROR "QUIC support is missing in OpenSSL/boringssl. Try setting -DOPENSSL_ROOT_DIR")
+#   endif()
+#   cmake_pop_check_state()
+# endfunction()
 
 option(USE_NGTCP2 "Use ngtcp2 and nghttp3 libraries for HTTP/3 support" OFF)
 if(USE_NGTCP2)
   if(USE_OPENSSL)
     find_package(NGTCP2 REQUIRED OpenSSL)
-    CheckQuicSupportInOpenSSL()
+    #CheckQuicSupportInOpenSSL()
   elseif(USE_GNUTLS)
     # TODO add GnuTLS support as vtls library.
     find_package(NGTCP2 REQUIRED GnuTLS)
@@ -717,17 +717,21 @@ if(ZLIB_FOUND)
   list(APPEND CMAKE_REQUIRED_INCLUDES ${ZLIB_INCLUDE_DIRS})
 endif()
 
-option(CURL_BROTLI "Set to ON to enable building curl with brotli support." OFF)
-set(HAVE_BROTLI OFF)
-if(CURL_BROTLI)
-  find_package(Brotli QUIET)
-  if(BROTLI_FOUND)
-    set(HAVE_BROTLI ON)
-    list(APPEND CURL_LIBS ${BROTLI_LIBRARIES})
-    include_directories(${BROTLI_INCLUDE_DIRS})
-    list(APPEND CMAKE_REQUIRED_INCLUDES ${BROTLI_INCLUDE_DIRS})
-  endif()
-endif()
+# option(CURL_BROTLI "Set to ON to enable building curl with brotli support." OFF)
+# set(HAVE_BROTLI OFF)
+# if(CURL_BROTLI)
+#   find_package(Brotli QUIET)
+#   if(BROTLI_FOUND)
+#     set(HAVE_BROTLI ON)
+#     list(APPEND CURL_LIBS ${BROTLI_LIBRARIES})
+#     include_directories(${BROTLI_INCLUDE_DIRS})
+#     list(APPEND CMAKE_REQUIRED_INCLUDES ${BROTLI_INCLUDE_DIRS})
+#   endif()
+# endif()
+
+# Force enable brotli
+set(HAVE_BROTLI ON)
+list(APPEND CURL_LIBS brotli)
 
 option(CURL_ZSTD "Set to ON to enable building curl with zstd support." OFF)
 set(HAVE_ZSTD OFF)
diff --git a/src/curl.rc b/src/curl.rc
index b0e328798..500ba5bd7 100644
--- a/src/curl.rc
+++ b/src/curl.rc
@@ -26,6 +26,8 @@
 
 LANGUAGE  0, 0
 
+ICONAPP ICON "curl.ico"
+
 #define RC_VERSION  CURL_VERSION_MAJOR, CURL_VERSION_MINOR, CURL_VERSION_PATCH, 0
 
 VS_VERSION_INFO VERSIONINFO
