diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1cdc524..b0d4125 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,11 +53,22 @@ project(libssh2 C)
 
 set(CMAKE_UNITY_BUILD_BATCH_SIZE 32)
 
+if(MSVC)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY_DEFAULT "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -DUNICODE=1 -D_UNICODE=1 -wd26812")
+  set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -permissive- -Zc:__cplusplus -DUNICODE=1 -D_UNICODE=1 -wd26812"
+  )
+  set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} -c65001")
+endif(MSVC)
+
+set(CMAKE_C_STANDARD 17)
+
 option(BUILD_STATIC_LIBS "Build Static Libraries" ON)
 add_feature_info("Static library" BUILD_STATIC_LIBS
   "creating libssh2 static library")
 
-option(BUILD_SHARED_LIBS "Build Shared Libraries" ON)
+option(BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
 add_feature_info("Shared library" BUILD_SHARED_LIBS
   "creating libssh2 shared library (.so/.dll)")
 
@@ -110,8 +121,8 @@ else()
   endif()
 endif()
 
-option(BUILD_EXAMPLES "Build libssh2 examples" ON)
-option(BUILD_TESTING "Build libssh2 test suite" ON)
+option(BUILD_EXAMPLES "Build libssh2 examples" OFF)
+option(BUILD_TESTING "Build libssh2 test suite" OFF)
 
 if(NOT BUILD_STATIC_LIBS AND NOT BUILD_SHARED_LIBS)
   set(BUILD_STATIC_LIBS ON)
