diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4161655..59a1243 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -156,6 +156,16 @@ set(LIBSSH2_LIBS "")
 set(LIBSSH2_LIBDIRS "")
 set(LIBSSH2_PC_REQUIRES_PRIVATE "")
 
+ 
+if(MSVC)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY_DEFAULT "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -DUNICODE=1 -D_UNICODE=1 -wd26812")
+  set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -permissive- -Zc:__cplusplus -DUNICODE=1 -D_UNICODE=1 -wd26812"
+  )
+  set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} -c65001")
+endif(MSVC)
+
 # Add socket libraries
 if(WIN32)
   list(APPEND LIBSSH2_LIBS_SOCKET "ws2_32")
