diff --git a/CMakeLists.txt b/CMakeLists.txt
index 813884a..f8a3801 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,15 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     "None" "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
+if(MSVC)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY_DEFAULT "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -DUNICODE=1 -D_UNICODE=1 -wd26812")
+  set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -permissive- -Zc:__cplusplus -DUNICODE=1 -D_UNICODE=1 -wd26812"
+  )
+  set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} -c65001")
+endif(MSVC)
+
 include(GNUInstallDirs)
 include(CMakeDependentOption)
 
@@ -150,10 +159,11 @@ if(ENABLE_OPENSSL AND OPENSSL_FOUND)
   if(WIN32)
     set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES}" "ws2_32" "bcrypt")
   endif()
-  check_symbol_exists(SSL_provide_quic_data "openssl/ssl.h" HAVE_SSL_PROVIDE_QUIC_DATA)
-  if(NOT HAVE_SSL_PROVIDE_QUIC_DATA)
-    message(WARNING "Disabling OpenSSL due to lack of QUIC support in ${OPENSSL_LIBRARIES}")
-  endif()
+  # check_symbol_exists(SSL_provide_quic_data "openssl/ssl.h" HAVE_SSL_PROVIDE_QUIC_DATA)
+  # if(NOT HAVE_SSL_PROVIDE_QUIC_DATA)
+  #   message(WARNING "Disabling OpenSSL due to lack of QUIC support in ${OPENSSL_LIBRARIES}")
+  # endif()
+  set(HAVE_SSL_PROVIDE_QUIC_DATA ON)
   cmake_pop_check_state()
 endif()
 if(ENABLE_OPENSSL AND HAVE_SSL_PROVIDE_QUIC_DATA)
