diff --git a/CMakeLists.txt b/CMakeLists.txt
index 80843c3..353c071 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -82,6 +82,15 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     "None" "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
+if(MSVC)
+  set(CMAKE_MSVC_RUNTIME_LIBRARY_DEFAULT "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -DUNICODE=1 -D_UNICODE=1 -wd26812")
+  set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS=1 -utf-8 -W3 -permissive- -Zc:__cplusplus -DUNICODE=1 -D_UNICODE=1 -wd26812"
+  )
+  set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} -c65001")
+endif(MSVC)
+
 include(GNUInstallDirs)
 
 include(CMakeOptions.txt)
diff --git a/lib/ngtcp2_ringbuf.c b/lib/ngtcp2_ringbuf.c
index 4ad67ad..ce4cf81 100644
--- a/lib/ngtcp2_ringbuf.c
+++ b/lib/ngtcp2_ringbuf.c
@@ -33,7 +33,7 @@
 
 #if defined(_MSC_VER) && !defined(__clang__) &&                                \
     (defined(_M_ARM) || defined(_M_ARM64))
-unsigned int __popcnt(unsigned int x) {
+static __forceinline unsigned int __popcnt(unsigned int x) {
   unsigned int c = 0;
   for (; x; ++c) {
     x &= x - 1;
